### ISSUE TITLE
Strider (2014) has lighting / texture issues.

### ISSUE THREAD

**Issue by strycore** [doitsujin/dxvk#444]
Capcom's Strider has some rendering issues, showing some areas in black. Running the game without DXVK shows similar glitches (with OpenGL, some effects aren't rendering at all). 

![screenshot from 2018-06-22 15-47-53](https://user-images.githubusercontent.com/280336/41802678-65f98f88-7637-11e8-8e80-bf695ed54bdf.jpg)

![screenshot from 2018-06-22 15-47-57](https://user-images.githubusercontent.com/280336/41802680-68fd1aec-7637-11e8-9f4c-94cc3d196366.jpg)

### Software information
Strider (2014) on Steam, all graphical settings apply.

### System information
- GPU: GTX970
- Driver: Nvidia 396.24.02
- Wine version: 3.10 Staging
- DXVK version: 0.60

### Apitrace file(s)
- https://mega.nz/#!Op5kRIIQ!tTESNRe8EB8WztaO4culVZwWK4BaXMeBQadLm0AaWUQ

### Log files
- d3d11.log: [StriderX64_r_d3d11.log](https://github.com/doitsujin/dxvk/files/2129387/StriderX64_r_d3d11.log)
- dxgi.log: [StriderX64_r_dxgi.log](https://github.com/doitsujin/dxvk/files/2129388/StriderX64_r_dxgi.log)

**Comment by doitsujin** [comment]
Can you point out what is wrong in your screenshot, or post a screenshot of what it should look like?

**Comment by strycore** [comment]
Sure, here is what it's supposed to look like on PC:

![screenshot from 2018-06-22 16-36-30](https://user-images.githubusercontent.com/280336/41803109-a79d1c40-763a-11e8-8fad-d167fe5002f0.jpg)

Here is the Let's Play this is from: https://www.youtube.com/watch?v=APSWtoOIG08&index=6&list=PLqUI_YQcw3DY7rfSq_-orAwH2CBNIeDIU

**Comment by strycore** [comment]
Here is a trace produced on Windows 10, with Direct3D: https://mega.nz/#!it4GCATC!3XnWl6Gps9vuIM80CSAGQP5sKqN9AAyhJXGWFUY-r2o

I tried replaying the trace and it looked fine, no glitches there.

**Comment by strycore** [comment]
Good news, I've managed to get the game to render without issues!
I've edited the game's config file in Steam/userdata/10042493/235210/local/vars_pc.cfg to change 
`video_multithreadedRendering` to false.

Based on that, it seems that the original issue is related to a multi-threading issue, which now makes more sense.

**Comment by corvusd** [comment]
First at all, thanks to the devs and community for this awesome effort! 
In my case not work. I change de config in file, but the game reset in each start. If put only read file, same situation. If possible Multithreaded Rendering a feature to implement in dxvk? Thanks community! :)

**Comment by doitsujin** [comment]
Multithreaded Rendering is an in-game option, not a D3D11 feature.

**Comment by corvusd** [comment]
> Multithreaded Rendering is an in-game option, not a D3D11 feature.

I see, and this bug or problem, because happens? Driver(I have nvidia gtx1060) wine or wine-staging part?

**Comment by corvusd** [comment]
Well, only I want to help well. I mean, if not report here... where is the best place to report this issue to help to find some solution. Is a very "silly" bug for a great game, that in all the rest work perfectly, even in the part of dxvk that give extraordinary support to run dx11 at superb performance. Thanks in advance.

**Comment by corvusd** [comment]
Well, I try to play in proton only. Then now I think check in native wine-staging session and the workaround work well. Change de .cfg to false and Render Ok. I think this issue is for report in proton.
For the issue of the Multithreaded Rendering I don’t known where, any ideas?
Thanks again and apologizes for the mistake to report. Greetings.

**Comment by K0bin** [comment]
I don't really understand that last comment. What are you trying to say?

**Comment by K0bin** [comment]
It did seem like a game bug but as I said, take what I said with a grain of salt and please don't open additional issues about this before confirming it.

**Comment by corvusd** [comment]
> It did seem like a game bug but as I said, take what I said with a grain of salt and please don't open additional issues about this before confirming it.

Okay, I did not understand that it was a bug of the game itself. Rather a feature that fails to reproduce the same.

**Comment by doitsujin** [comment]
@K0bin could you test this with `d3d11.dcSingleUseMode = False`?

**Comment by K0bin** [comment]
I don't own the game. @strycore can you do that?

**Comment by massatt212** [comment]
![screenshot from 2019-01-14 11-40-39](https://user-images.githubusercontent.com/43518479/51122903-4f823000-17f1-11e9-8155-6325288546a4.png)
same problem
if i did it wrong tell me

**Comment by K0bin** [comment]
Can you upload the logs? That way we could see if DXVK has applied your config file.

**Comment by massatt212** [comment]
[StriderX64_r_d3d11.log](https://github.com/doitsujin/dxvk/files/2755884/StriderX64_r_d3d11.log)
[StriderX64_r_dxgi.log](https://github.com/doitsujin/dxvk/files/2755885/StriderX64_r_dxgi.log)

**Comment by K0bin** [comment]
The logs dont mention the config file so you must have done something wrong.

**Comment by massatt212** [comment]
make the conf file incase thats where im going wrong and ill download yours and test again

**Comment by massatt212** [comment]
it worked using user_settings.py under proton
the game runs fine
with d3d11.dcSingleUseMode = False

**Comment by massatt212** [comment]
[StriderX64_r_d3d11.log](https://github.com/doitsujin/dxvk/files/2756209/StriderX64_r_d3d11.log)
[StriderX64_r_dxgi.log](https://github.com/doitsujin/dxvk/files/2756210/StriderX64_r_dxgi.log)
this one detected the dxvk.conf file but it didnt work

**Comment by K0bin** [comment]
It detected the config file but there's no mention of d3d11.dcSingleUseMode = False

**Comment by massatt212** [comment]
Game: StriderX64_r.exe
info:  DXVK: v0.95
info:  Found config file: dxvk.conf
info:  Effective configuration:
info:    dxgi.deferSurfaceCreation = False

**Comment by massatt212** [comment]
[StriderX64_r_d3d11.log](https://github.com/doitsujin/dxvk/files/2756298/StriderX64_r_d3d11.log)
[StriderX64_r_dxgi.log](https://github.com/doitsujin/dxvk/files/2756299/StriderX64_r_dxgi.log
sorry prob uploaded me experiment

**Comment by K0bin** [comment]
That last one has the config option. It didnt work, did it?

**Comment by massatt212** [comment]
nope did not fix anything
even when i do the video_multithreadedRendering to false doesnt work

**Comment by massatt212** [comment]
Strider Works Fine On Windows Via DXVK
I think its a Graphical Problem similar to Killer Instinct Missing Textures on the loading screen on Linux Also
but the strange thing is i got it working on Ubuntu earlier by mistake doing random shit so it maybe a linux driver problem or Wine
https://youtu.be/MqZ5bQa4zo8

**Comment by doitsujin** [comment]
Hm, that's interesting. I think we can rule out a driver issue since it is known to happen on both RADV and Nvidia, but that would indicate that there's a thread-safety issue elsewhere. Could you post the full output with `PROTON_LOG=1`?

**Comment by massatt212** [comment]
[steam-235210.log](https://github.com/doitsujin/dxvk/files/2757019/steam-235210.log)
Is this it ?

**Comment by doitsujin** [comment]
Yes, thanks. Can't find anything that would hint at a missing function in wine or anything like that, though, so not sure what's going on.
